test/app.test.ts

```diff
@@ -0,0 +1,156 @@
+import http = require("http");
+import request = require("supertest");
+import should from "should";
+import { app } from "../src/app";
+import config from "../src/config";
+let server: http.Server;
+import async from "async";
+import { ErrorCode } from "../src/helpers";
+
+describe("app e2e测试", () => {
+    before((done) => {
+        server = app.listen(config.PORT, () => {
+            console.log(`测试服务器已启动：http://127.0.0.1:${config.PORT}`);
+            done();
+        });
+    });
+    after((done) => {
+        server.close(done);
+    });
+
+    // user
+
+    it("Create User should be success", (done) => {
+        request(server)
+            .post("/user")
+            .send({ name: "john" })
+            .expect(200, (err, res) => {
+                if (err) {
+                    done(err);
+                    return;
+                }
+                should(res.body.statusCode === 200).ok();
+                done();
+            });
+    });
+
+    it("Create User without name be blocked", (done) => {
+        request(server)
+            .post("/user")
+            .send({ name: "" })
+            .expect(500, (err, res) => {
```

Comment:
hmm, why do you think 500 should be a fit for this case?

Reply:
It's also ok with 200 response and returns a custom error code to make the frontend how to handle this.
In this case, an empty name will be blocked by `koa-swagger-decorator validation` and it will throw an error to `error` middleware and it will lead to status 500.